File: python/examples/jobs/02_job_management_http.py
"""
Example: Job management using HTTP API

This example demonstrates:
- GET /v1/crawl/jobs/{id} - Get job details
- DELETE /v1/crawl/jobs/{id} - Cancel/delete job
- GET /v1/crawl/jobs/{id}/download - Get download URL
"""

import requests
import json

API_KEY = "YOUR_API_KEY"
BASE_URL = "https://api.crawl4ai.com"

headers = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json"
}

# Create a test job
print("=== Creating Test Job ===")
response = requests.post(
    f"{BASE_URL}/v1/crawl/async",
    headers=headers,
    json={
        "urls": ["https://example.com", "https://example.org"],
        "priority": 5
    }
)
response.raise_for_status()
job = response.json()
job_id = job['job_id']
print(f"Created job: {job_id}")

# Get job details
print("\n=== Get Job Details ===")
response = requests.get(
    f"{BASE_URL}/v1/crawl/jobs/{job_id}",
    headers=headers
)
job_details = response.json()
print(f"Status: {job_details['status']}")
print(f"URLs: {job_details['urls']}")

# Cancel job (keep results)
print("\n=== Cancel Job (Keep Results) ===")
response = requests.delete(
    f"{BASE_URL}/v1/crawl/jobs/{job_id}",
    headers=headers,
    params={"delete_results": "false"}
)
cancelled = response.json()
print(f"Status: {cancelled['status']}")

# Create and delete completely
print("\n=== Cancel + Delete Results ===")
response = requests.post(
    f"{BASE_URL}/v1/crawl/async",
    headers=headers,
    json={"urls": ["https://example.com"]}
)
job2_id = response.json()['job_id']
print(f"Created job: {job2_id}")

response = requests.delete(
    f"{BASE_URL}/v1/crawl/jobs/{job2_id}",
    headers=headers,
    params={"delete_results": "true"}
)
print(f"Deleted: {response.json()['status']}")

# Get download URL
print("\n=== Get Download URL ===")
try:
    # Find a completed job
    response = requests.get(
        f"{BASE_URL}/v1/crawl/jobs",
        headers=headers,
        params={"status": "completed", "limit": 1}
    )
    jobs = response.json()

    if jobs['jobs']:
        completed_job_id = jobs['jobs'][0]['job_id']
        response = requests.get(
            f"{BASE_URL}/v1/crawl/jobs/{completed_job_id}/download",
            headers=headers,
            params={"expires_in": 3600}
        )
        download_data = response.json()
        print(f"Download URL: {download_data['url'][:100]}...")
        print("URL expires in 3600 seconds (1 hour)")
    else:
        print("No completed jobs found")
except Exception as e:
    print(f"Error: {e}")

File: nodejs/examples/jobs/02_job_management_http.ts
#!/usr/bin/env npx ts-node
/**
 * Example: Job management using HTTP API
 *
 * This example demonstrates:
 * - GET /v1/crawl/jobs/{id} - Get job details
 * - DELETE /v1/crawl/jobs/{id} - Cancel/delete job
 * - GET /v1/crawl/jobs/{id}/download - Get download URL
 *
 * Usage:
 *     npx ts-node 02_job_management_http.ts
 */

// Configuration
const API_KEY = 'YOUR_API_KEY';
const BASE_URL = 'https://api.crawl4ai.com';

const headers = {
  Authorization: `Bearer ${API_KEY}`,
  'Content-Type': 'application/json',
};

async function main(): Promise<void> {
  // Create a test job
  console.log('=== Creating Test Job ===');
  let response = await fetch(`${BASE_URL}/v1/crawl/async`, {
    method: 'POST',
    headers,
    body: JSON.stringify({
      urls: ['https://example.com', 'https://example.org'],
      priority: 5,
    }),
  });

  const job = await response.json();
  const jobId = job.job_id;
  console.log(`Created job: ${jobId}`);

  // Get job details
  console.log('\n=== Get Job Details ===');
  response = await fetch(`${BASE_URL}/v1/crawl/jobs/${jobId}`, { headers });
  const jobDetails = await response.json();
  console.log(`Status: ${jobDetails.status}`);
  console.log(`URLs: ${jobDetails.urls?.join(', ')}`);

  // Cancel job (keep results)
  console.log('\n=== Cancel Job (Keep Results) ===');
  response = await fetch(`${BASE_URL}/v1/crawl/jobs/${jobId}?delete_results=false`, {
    method: 'DELETE',
    headers,
  });
  const cancelled = await response.json();
  console.log(`Status: ${cancelled.status}`);

  // Create and delete completely
  console.log('\n=== Cancel + Delete Results ===');
  response = await fetch(`${BASE_URL}/v1/crawl/async`, {
    method: 'POST',
    headers,
    body: JSON.stringify({ urls: ['https://example.com'] }),
  });
  const job2Id = (await response.json()).job_id;
  console.log(`Created job: ${job2Id}`);

  response = await fetch(`${BASE_URL}/v1/crawl/jobs/${job2Id}?delete_results=true`, {
    method: 'DELETE',
    headers,
  });
  console.log(`Deleted: ${(await response.json()).status}`);

  // Get download URL
  console.log('\n=== Get Download URL ===');
  try {
    // Find a completed job
    response = await fetch(`${BASE_URL}/v1/crawl/jobs?status=completed&limit=1`, { headers });
    const jobs = await response.json();

    if (jobs.jobs?.length) {
      const completedJobId = jobs.jobs[0].job_id;
      response = await fetch(`${BASE_URL}/v1/crawl/jobs/${completedJobId}/download?expires_in=3600`, {
        headers,
      });
      const downloadData = await response.json();
      console.log(`Download URL: ${downloadData.url?.slice(0, 100)}...`);
      console.log('URL expires in 3600 seconds (1 hour)');
    } else {
      console.log('No completed jobs found');
    }
  } catch (e) {
    console.log(`Error: ${e}`);
  }
}

main().catch(console.error);
